pythermalcomfort/charts/heatmap.py

```diff
@@ -0,0 +1,215 @@
+        'DateTime' (Date and time), 'Hour' (Hour of the day), 'Month Name' (Month name), 'Day' (Day of the month),
+        and the variable to be plotted.
+
+    var : str
+        The variable to be plotted.
+
+    global_local : str
+        The scope of the data. It can be 'global' or 'local'.
+
+    si_ip : str
+        The unit system to use. It can be 'si' or 'ip'.
+
+    show_summary : bool, optional
+        If True, a summary of the data is shown. Default is False.
+
+    Returns
+    -------
+    fig : graph_objects.Figure
+        Plotly figure object.
+
+    """
+
+    var_name = index_mapping_dictionary[var]["name"]
+    var_unit = index_mapping_dictionary[var][si_ip]["unit"]
+    var_range = index_mapping_dictionary[var][si_ip]["range"]
+    var_color = index_mapping_dictionary[var]["colors"]
+
+    if global_local == "global":
+        range_z = var_range
+    else:
+        data_max = 5 * ceil(df[var].max() / 5)
+        data_min = 5 * floor(df[var].min() / 5)
+        range_z = [data_min, data_max]
+
+    if not show_summary:
+        fig = go.Figure(
+            go.Heatmap(
+                y=df["Hour"] + 0.5,  # Offset hour for better display
+                x=df["DateTime"].dt.date,
+                z=df[var],
+                zmin=range_z[0],
+                zmax=range_z[1],
+                customdata=np.stack(
+                    (df["Month Name"], df["Day"], df["Hour"], df["Hour"] + 1), axis=-1
+                ),
+                hovertemplate=(
+                    "<b>"
+                    + var
+                    + ": %{z:.2f} "
+                    + var_unit
+                    + "</b><br>Month: %{customdata[0]}<br>Day: %{customdata[1]}<br>Hour:"
+                    " %{customdata[2]}-%{customdata[3]}<br>"
+                ),
+                name="",
+                colorscale=var_color,
+                colorbar=dict(
+                    title=dict(
+                        text=f"{var_name}" + " in " + f"{var_unit} ", side="right"
+                    ),
+                    orientation="h",
+                    thickness=20,
+                    x=0,
+                    xref="paper",
+                    xanchor="left",
+                    y=0.85,
+                    yref="container",
+                    yanchor="bottom",
+                    ticks="inside",
+                    ticklabelposition="outside top",
+                ),
+            )
+        )
+
+        fig.update_yaxes(
+            title_text="Hour", mirror=True, range=[0, 24], zerolinewidth=1, dtick=2
+        )
+        fig.update_xaxes(
+            dtick="M1",
+            tickformat="%b",
+            ticklabelmode="period",
+            title_text="Day",
+            mirror=True,
+            zeroline=False,
+        )
+
+        fig.update_layout(margin=tight_margins, width=800, height=500)
+
+    else:
+        cat_color = index_mapping_dictionary[var]["colors_categories"]
```

Comment:
Same as the previous comment about the string. As previously mentioned, the categories should be extracted using a function within data class.